src/lib/AbstractSyntax.hs

```diff
@@ -391,6 +395,7 @@ effect (Binary Juxtapose (Identifier "State") (Identifier h)) =
   return $ RWSEffect State $ (Just $ fromString h)
 effect (Identifier "Except") = return ExceptionEffect
 effect (Identifier "IO") = return IOEffect
+effect (Identifier effName) = return $ UserEffect (fromString effName)
 effect _ = throw SyntaxErr "Unexpected effect form; expected one of `Read h`, `Accum h`, `State h`, `Except`, or `IO`."
```

Comment:
Please update the error message here (", IO, or a name of a user-defined effect").